### PR TITLE
Feat(generator): Make RestrictedVector global

### PR DESCRIPTION
**Pull request by maksk7**

### CODE REVIEW

File: packages/astroport/src/restricted_vector.rs [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+    pub fn get_last(&self, key: &K) -> StdResult<V> {
+        self.0
+            .last()
+            .filter(|(k, _)| k == key)
+            .map(|(_, v)| v)
+            .cloned()
+            .ok_or_else(|| StdError::generic_err(format!("Key {} not found", key)))
+    }
+
+    pub fn update(&mut self, key: K, value: V) -> StdResult<V> {
```
Comment (epanchee) [inline packages/astroport/src/restricted_vector.rs:38]: I think we may just borrow 'key' here like this ```..., key: &K, ... ```   
Thus we can escape cloning of 'key' every time we call this function.

File: packages/astroport/src/restricted_vector.rs [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+    }
+}
+
+impl Increaseable for Uint128 {
+    fn increase(self, new: Uint128) -> StdResult<Uint128> {
+        self.checked_add(new).map_err(Into::into)
+    }
+}
+
+impl<K, V> Default for RestrictedVector<K, V> {
```
Comment (epanchee) [inline packages/astroport/src/restricted_vector.rs:82]: I believe we can use ```#[dervie(Default)]``` instead of manual implementation

Comment (maksk7) [reply packages/astroport/src/restricted_vector.rs:82]: the trait `Default` is not implemented for `Addr`. So in this case I implemented it manually.

File: packages/astroport/src/restricted_vector.rs (hunk 1 above)
Comment (maksk7) [reply packages/astroport/src/restricted_vector.rs]: Ok, done

File: packages/astroport/src/restricted_vector.rs (hunk 2 above)
Comment (epanchee) [reply packages/astroport/src/restricted_vector.rs:82]: Ok, got it

**Review by epanchee — APPROVED** [review]

**Review by P-Yevhenii — APPROVED** [review]

**Review by ShadoySV — APPROVED** [review]

**Pull request merged**